pennylane/measurements/counts.py

```diff
@@ -146,9 +145,6 @@ def circuit():
 
         return CountsMP(obs=op, all_outcomes=all_outcomes)
 
-    if op is not None and not op.is_hermitian:  # None type is also allowed for op
-        warnings.warn(f"{op.name} might not be hermitian.")
-
     if wires is not None:
```

Comment:
Would it also be an option to do this?
```suggestion
    if op is not None and not qml.math.is_abstract(op.data) and not op.is_hermitian:  # None type is also allowed for op
        warnings.warn(f"{op.name} might not be hermitian.")
    if wires is not None:
```
This could keep the test/warning JIT-friendly.

I'm not sure how important these warnings are, though, so maybe it's just nicer to remove them :)

Reply:
This was just a guard we put in place when we started allowing non-`Observable`'s through.

I've seen this actually being more expensive than I anticipated, so I was actually wanting this to be removed for performance reasons anyway.

Reply:
Would be a possibility, yeah 🤔 I personally dont have a strong take on this. Any opinions?

Reply:
cc @isaacdevlugt @rmoyard

Reply:
I agree with Christina, imo it is up to the user to give valid programs.

Reply:
Am okay with removing the warning altogether :). I've said this elsewhere, but I liken this to how we can pass imaginary rotation angles to rotation gates. Non-physical result, but no warning. A feature, not a bug! Should hopefully be obvious to users that that's what's going on if it happens by accident 🙏

Reply:
Great, gonna merge now :)